src/imp/libc/net/syscalls.rs

```diff
@@ -400,9 +400,15 @@ pub(crate) mod sockopt {
     #[inline]
     fn getsockopt<T>(fd: BorrowedFd<'_>, level: i32, optname: i32) -> io::Result<T> {
```

Comment:
I think I'd say we shouldn't support any Rust type for `T` here but rather have `getsockopt_int` and `getsockopt_ptr` or so?

Reply:
The `T` could also be types like `struct linger` or `struct timespec`, or typedefs which may be 32-bit or 64-bit depending on the platform, so it's fairly handy to make it generic.

[Rust's standard library does this too](https://github.com/rust-lang/rust/blob/master/library/std/src/sys_common/net.rs#L79). In looking that up, I noticed that Rust uses `T: Copy`, which should help catch some kinds of misuse, so I've added that here too.

Reply:
It'd be nice if there was some way to express though `T: #[repr(C)]` or so.  Hmm, seems like e.g. [Layout](https://doc.rust-lang.org/std/alloc/struct.Layout.html) could have something like a `const fn is_c()` or something?

Anyways, still LGTM.